pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -997,6 +1012,14 @@ type ScrapeEndpoint struct {
 	// instance, or __address__) are not permitted. The labelmap action is not permitted
 	// in general.
 	MetricRelabeling []RelabelingRule `json:"metricRelabeling,omitempty"`
+	// Prometheus HTTP client configuration.
+	HTTPClientConfig `json:",inline"`
+}
+
+// HTTPClientConfig stores HTTP-client configurations.
+type HTTPClientConfig struct {
```

Comment:
I think this might be different to what was in the design doc so far (https://docs.google.com/document/d/16erW70zg-F8CE0Kf3GHMASRsdNnMUNfdCI5YvfnNXxw/edit?resourcekey=0-N8ZdIwLMf7uNdm1XmckMaQ#heading=h.sfylv0d4e4lc). Let's make it consistent as it is the change we have to be careful here (:

Reply:
Maybe I took the design outline too literally?